pkg/common/controlplaneendpoint/virtualip/kubevip_test.go

```diff
@@ -0,0 +1,192 @@
+)
+
+func Test_GetFile(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name                     string
+		controlPlaneEndpointSpec v1alpha1.ControlPlaneEndpointSpec
+		configMap                *corev1.ConfigMap
+		expectedContent          string
+		expectedErr              error
+	}{
+		{
+			name: "should return templated data with both host and port",
+			controlPlaneEndpointSpec: v1alpha1.ControlPlaneEndpointSpec{
+				Host: "10.20.100.10",
+				Port: 6443,
+			},
+			configMap: &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "default-kube-vip-template",
+					Namespace: "default",
+				},
+				Data: map[string]string{
+					"data": validKubeVIPTemplate,
+				},
+			},
+			expectedContent: expectedKubeVIPPod,
+		},
+	}
+
+	for idx := range tests {
+		tt := tests[idx] // Capture range variable
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			fakeClient := fake.NewClientBuilder().WithObjects(tt.configMap).Build()
+
+			provider := kubeVIPFromConfigMapProvider{
+				client:       fakeClient,
+				configMapKey: client.ObjectKeyFromObject(tt.configMap),
+			}
+
+			file, err := provider.GetFile(context.TODO(), tt.controlPlaneEndpointSpec)
+			require.Equal(t, tt.expectedErr, err)
+			assert.Equal(t, tt.expectedContent, file.Content)
+			assert.NotEmpty(t, file.Path)
+			assert.NotEmpty(t, file.Owner)
+			assert.NotEmpty(t, file.Permissions)
+		})
+	}
+}
+
+func Test_getTemplateFromConfigMap(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name         string
+		configMap    *corev1.ConfigMap
+		expectedData string
+		expectedErr  error
+	}{
+		{
+			name: "should return data from the only key",
+			configMap: &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "default-kube-vip-template",
+					Namespace: "default",
+				},
+				Data: map[string]string{
+					"data": "kube-vip-template",
+				},
+			},
+			expectedData: "kube-vip-template",
+		},
+		{
+			name: "should fail with multipleKeysError",
+			configMap: &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "default-kube-vip-template",
+					Namespace: "default",
+				},
+				Data: map[string]string{
+					"data":           "kube-vip-template",
+					"unexpected-key": "unexpected-value",
+				},
+			},
+			expectedErr: multipleKeysError{
+				configMapKey: client.ObjectKey{
+					Name:      "default-kube-vip-template",
+					Namespace: "default",
+				},
+			},
+		},
+		{
+			name: "should fail with emptyValuesError",
+			configMap: &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "default-kube-vip-template",
+					Namespace: "default",
+				},
+				Data: map[string]string{
+					"data": "",
+				},
+			},
+			expectedErr: emptyValuesError{
+				configMapKey: client.ObjectKey{
+					Name:      "default-kube-vip-template",
+					Namespace: "default",
+				},
+			},
+		},
+	}
+
+	for idx := range tests {
+		tt := tests[idx] // Capture range variable
```

Comment:
No need to do this any more with go 1.22 🎉 

```suggestion
	for _, tt := range tests {
```

Reply:
thanks, good to know!